gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/BucketRegion.java

```diff
@@ -1295,20 +1295,23 @@ private boolean writeLockEnabled() {
     }
   }
 
-  private volatile Boolean rowBuffer = false;
+  private volatile Boolean rowBuffer = null;
 
   public boolean isRowBuffer() {
     final Boolean rowBuffer = this.rowBuffer;
-    if (rowBuffer || this.getName().toUpperCase().endsWith(StoreCallbacks.SHADOW_TABLE_SUFFIX)) {
+    if (rowBuffer != null) {
       return rowBuffer;
     }
-    boolean isRowBuffer = false;
-    List<PartitionedRegion> childRegions = ColocationHelper.getColocatedChildRegions(this.getPartitionedRegion());
+    List<PartitionedRegion> childRegions = ColocationHelper.getColocatedChildRegions(
+        this.getPartitionedRegion());
     for (PartitionedRegion pr : childRegions) {
-      isRowBuffer |= pr.getName().toUpperCase().endsWith(StoreCallbacks.SHADOW_TABLE_SUFFIX);
+      if (pr.getName().toUpperCase().endsWith(StoreCallbacks.SHADOW_TABLE_SUFFIX)) {
+        this.rowBuffer = true;
+        return true;
+      }
     }
-    this.rowBuffer = isRowBuffer;
-    return isRowBuffer;
+    this.rowBuffer = false;
+    return false;
   }
 
 
```

Comment:
Store precheckin does not have rowBuffer so I don't expect it to show anything. Will back this out to be safe.